sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -122,6 +122,7 @@ abstract class Optimizer(catalogManager: CatalogManager)
       Batch("Operator Optimization before Inferring Filters", fixedPoint,
         rulesWithoutInferFiltersFromConstraints: _*) ::
       Batch("Infer Filters", Once,
+        InferFiltersFromGenerate,
```

Comment:
Why did you put this rule only in this batch? https://github.com/apache/spark/pull/29092/files#diff-a636a87d8843eeccca90140be91d4fafR82

Reply:
I'm sorry, I don't get the question.

Reply:
Ah, ok. Looks okay as it is. Since `operatorOptimizationRuleSet` has `InferFiltersFromConstraints`, I thought we'd better to put `InferFiltersFromConstraints`/`InferFiltersFromGenerate` there together.

Reply:
Actually the `InferFiltersFromConstraints` is allways filtered out from the `operatorOptimizationRuleSet`:
https://github.com/apache/spark/blob/5ce321dc80a699fa525ca5b69bf2c28e10f8a12a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L120-L123

I think it is very misleading , perhaps we could remove it from there and avoid further confusion?

It was asked in the original PR, put looks like it was forgotten:
https://github.com/apache/spark/pull/19149/files#r157777935

Reply:
Yea, but I also don't know why. cc: @cloud-fan

Reply:
I don't quite remember, maybe we can remove it.

Reply:
I went ahead and removed it from the `operatorOptimizationRuleSet`. All the tests are still passing as was expected.